### PR TITLE
Use requirements.txt to specify dependencies. Delete egg-info

### LINKED ISSUE #1648: List python runtime dependencies in requirements.txt as opposed to setup.py
- [ ] I have searched to see if a similar issue already exists.

**Is your feature request related to a problem? Please describe.**  
As a gradio developer, I think it's frustrating that the `gradio-egg.info` folder is tracked in version control. Whenever I build gradio locally, there are changes in the `gradio-egg.info` directory that are not related to my PR.

I see that `gradio-egg.info` is only included in the repo so that we can
1. cache the env in circleci 
2. install all dependencies in the `install_test_requirements.sh` script

**Describe the solution you'd like**  
I'd like to specify all dependencies in a `requirements.txt` file. That will allow us to still cache the circle-ci env. I don't think we need to install the gradio dependencies again in `install_test_requirements.sh` since that's done in `install_gradio.sh` but we can still install from `requirements.txt` if necessary. 

This will also allow us to remove `gradio-egg.info` from version control. 

**Additional context**

### PR DESCRIPTION
**Pull request by freddyaboulton**
# Description

Fixes #1648

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My code follows the style guidelines of this project
- [x] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: gradio/__init__.py [hunk 1]
```diff
@@ -67,5 +67,5 @@
     Webcam,
 )
 
-current_pkg_version = pkg_resources.require("gradio")[0].version
+current_pkg_version = version("gradio")
```
Comment (freddyaboulton) [inline gradio/__init__.py:70]: Use pkg_resources is discouraged by setuptools

image

File: .circleci/config.yml [hunk 2]
```diff
@@ -23,13 +23,14 @@ jobs:
           command: |
             python3 -m venv venv
             . venv/bin/activate
+            bash scripts/install_gradio.sh
```
Comment (freddyaboulton) [inline .circleci/config.yml:26]: So that gradio is installed in circleci worker. We were doing this implicitly by having the egg-info in version control.

**Review by aliabid94 — APPROVED** [review]
Beautiful!

**Review by FarukOzderim — APPROVED** [review]
This looks great! It was definitely a headache, Didn't know that we could remove gradio.egg, thought it was necessary as a pip package.

Where do we keep the egg info now? Was it not necessary at all?

**Review by dawoodkhan82 — APPROVED** [review]

**Comment by abidlabs** [comment]
Egg info is not necessary as eggs are pretty much an abandoned format: https://stackoverflow.com/a/68897551/5209347

**Pull request merged**